fuse_core/include/fuse_core/parameter.hpp

```diff
@@ -124,6 +125,32 @@ T getParam(
   }
 }
 
+namespace detail {
+/** @brief Internal function for unit testing.
+ * @internal
+*/
+std::unordered_set<std::string>
+list_parameter_overrides_at_prefix(
+  const std::map<std::string, rclcpp::ParameterValue> & overrides,
+  std::string prefix, size_t depth);
+}  // namespace detail
+
+/**
+ * @brief Get pararameter overrides that exist at a given prefix.
+ * 
+ * Example:
+ *  Say the given parameter overrides are foo, foo.bar, foo.bar.baz, and foobar.baz
+ *  Given prefix "foo" and depth 0, this will return foo.bar and foo.bar.baz
+ *  Given prefix "foo" and depth 1, this will return foo.bar
+ * 
+ * @param[in] interfaces - The node interfaces used to get the parameter overrides
+ * @param[in] prefix - the name of a parameter 
+*/
+std::unordered_set<std::string>
+list_parameter_overrides_at_prefix(
+  node_interfaces::NodeInterfaces<node_interfaces::Parameters> interfaces,
+  std::string prefix, size_t depth = 0);
```

Comment:
Let's do this instead (I think it's a slightly more intuitive naming), so it doesn't get confused with "get **at** n-depth", but instead does "get **up-to** n-depth"
```suggestion
  std::string prefix, size_t max_depth = -1);
```

Reply:
Ah, I didn't even know about `depth` in `list_parameters` signature.

In order to avoid confusing the two APIs I renamed `list_parameter_overrides_at_prefix` to `list_parameter_override_prefixes` and removed the depth argument in 1343096. It now always behaves like it had been given a depth of `1` in the old implementation, which is all `fuse` needs.